reactor-core/src/main/java/reactor/core/publisher/FluxZip.java

```diff
@@ -506,7 +531,7 @@ public void onSubscribe(Subscription s) {
 		@Override
 		public void onNext(T t) {
 			if (done) {
-				Operators.onNextDropped(t, parent.currentContext());
+				Operators.onDiscard(t, parent.currentContext());
```

Comment:
`isBetween` checks that number of events is between 0 and 50%. Actually - number can be 0 as well as 100% since verification relies on technically random event with **finit** number of observations. 

If we puts that on clear statistic - both events have equal probability. However, that is true for **infinit** number of experiments. If we have **fixed** number of experiments, we have 0 guarantees to get exactly 50 as well no guarantee we always see less or more.

Thus, this test from statistical verification is invalid and gives nothing.

Reply:
Done in that case brings wrong impression. We have to remeber that upstream is flux and we set it as done once first element lands in the method. The other onNext call which may appear after cencellation must be discarded. Although i got your point. Will improve